inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java

```diff
@@ -213,6 +218,22 @@ private String submitJobBySavepoint(FlinkInfo flinkInfo, SavepointRestoreSetting
         }).filter(Objects::nonNull).collect(Collectors.toList());
 
         Configuration configuration = getFlinkConfiguration(flinkInfo.getEndpoint());
+        log.debug("flink info: {}", flinkInfo);
+        if (flinkConfig.getDynamicParallelismEnable()) {
+            flinkParallelismOptimizer.setMaximumMessagePerSecondPerCore(flinkConfig.getMaxMsgRatePerCore());
+            // get stream info list for auditing
+            int recommendedParallelism =
+                    flinkParallelismOptimizer.calculateRecommendedParallelism(flinkInfo.getInlongStreamInfoList());
+            // Ensure parallelism is at least the default value
+            if (recommendedParallelism < parallelism) {
+                recommendedParallelism = parallelism;
+            }
+            if (recommendedParallelism != parallelism) {
+                log.info("switched to recommended parallelism: {}", recommendedParallelism);
+                parallelism = recommendedParallelism;
+            }
+        }
+        log.info("current parallelism: {}", parallelism);
```

Comment:
Is it reasonable that when audit data cannot be obtained, the `recommendedParallelism` will be 0？

Reply:
Thanks a lot for your extremely rigorous review. However, I believe with the comparison in the red rectangle, the minimum parallelism is ensured, unless the parallelsim itself in `flink-sort-plugin.properties` is illegal.

![image](https://github.com/user-attachments/assets/ad39ce48-3d53-4767-98e5-4ecd4a911c48)

Reply:
OK